File: src/main/java/com/softwareplace/service/AuthorizationUserService.java
package com.softwareplace.service;

import java.util.HashMap;
import java.util.Map;

import javax.servlet.FilterChain;
import javax.servlet.http.HttpServletRequest;
import javax.servlet.http.HttpServletResponse;

import org.springframework.security.core.Authentication;

import com.softwareplace.model.RequestUser;
import com.softwareplace.model.UserData;

public interface AuthorizationUserService {

	UserData userData(RequestUser user);

	UserData userData(String authToken);

	long expirationTime();

	String secret();

	default void successfulAuthentication(HttpServletRequest request, HttpServletResponse response, FilterChain chain, Authentication authResult) {

	}

	default Map<String, Object> claims(HttpServletRequest httpServletRequest, UserData userData) {
		return new HashMap<>();
	}
}

File: src/main/java/com/softwareplace/service/AuthorizationUserService.kt
package com.softwareplace.service

import com.softwareplace.model.RequestUser
import com.softwareplace.model.UserData
import org.springframework.security.core.Authentication
import javax.servlet.FilterChain
import javax.servlet.http.HttpServletRequest
import javax.servlet.http.HttpServletResponse

interface AuthorizationUserService {

    fun userData(user: RequestUser): UserData?
    fun userData(authToken: String): UserData?
    fun expirationTime(): Long
    fun secret(): String
    fun successfulAuthentication(request: HttpServletRequest, response: HttpServletResponse, chain: FilterChain, authResult: Authentication) {}
    fun claims(httpServletRequest: HttpServletRequest?, userData: UserData): Map<String, Any> {
        return HashMap()
    }
}
